src/hotspot/share/opto/type.cpp

```diff
@@ -1789,264 +1697,122 @@ bool TypeInt::is_finite() const {
   return true;
 }
 
-//------------------------------dump2------------------------------------------
-// Dump TypeInt
-#ifndef PRODUCT
-static const char* intname(char* buf, size_t buf_size, jint n) {
-  if (n == min_jint)
-    return "min";
-  else if (n < min_jint + 10000)
-    os::snprintf_checked(buf, buf_size, "min+" INT32_FORMAT, n - min_jint);
-  else if (n == max_jint)
-    return "max";
-  else if (n > max_jint - 10000)
-    os::snprintf_checked(buf, buf_size, "max-" INT32_FORMAT, max_jint - n);
-  else
-    os::snprintf_checked(buf, buf_size, INT32_FORMAT, n);
-  return buf;
-}
-
-void TypeInt::dump2( Dict &d, uint depth, outputStream *st ) const {
-  char buf[40], buf2[40];
-  if (_lo == min_jint && _hi == max_jint)
-    st->print("int");
-  else if (is_con())
-    st->print("int:%s", intname(buf, sizeof(buf), get_con()));
-  else if (_lo == BOOL->_lo && _hi == BOOL->_hi)
-    st->print("bool");
-  else if (_lo == BYTE->_lo && _hi == BYTE->_hi)
-    st->print("byte");
-  else if (_lo == CHAR->_lo && _hi == CHAR->_hi)
-    st->print("char");
-  else if (_lo == SHORT->_lo && _hi == SHORT->_hi)
-    st->print("short");
-  else if (_hi == max_jint)
-    st->print("int:>=%s", intname(buf, sizeof(buf), _lo));
-  else if (_lo == min_jint)
-    st->print("int:<=%s", intname(buf, sizeof(buf), _hi));
-  else
-    st->print("int:%s..%s", intname(buf, sizeof(buf), _lo), intname(buf2, sizeof(buf2), _hi));
-
-  if (_widen != 0 && this != TypeInt::INT)
-    st->print(":%.*s", _widen, "wwww");
-}
-#endif
-
 //------------------------------singleton--------------------------------------
 // TRUE if Type is a singleton type, FALSE otherwise.   Singletons are simple
 // constants.
 bool TypeInt::singleton(void) const {
-  return _lo >= _hi;
+  return _lo == _hi;
 }
 
 bool TypeInt::empty(void) const {
-  return _lo > _hi;
+  return false;
```

Comment:
Oh, so it can now never be empty, why?

Reply:
A `TypeInt::try_make` that returns an empty type will return `Type::TOP`